### PR TITLE
Add bot-helpers package

### PR DESCRIPTION
**Pull request by timkinnane**
@RocketChat/core 

No issue number. This is a new feature nobody asked for, but I find very useful.

Provides methods for Hubots to query user collections. e.g. Loop through all online users or query a user's language setting before running translations.

See packages/rocketchat-bot-helpers/README.md for detailed usage.

### CODE REVIEW

**Comment by Sing-Li** [comment]
Thank you, @timkinnane !    Please make the package default "commented out" or disabled.

While very useful, they are prime DoS vectors for large deployments.  Making the package default disabled will allow those who understand the consequences to use it with discretion.   Thanks.

**Comment by timkinnane** [comment]
Thanks @Sing-Li, I wasn't sure about this one but thought I'd start with the PR anyway. I feel much more comfortable with that approach, will update and fix the codacy issues.

**Comment by timkinnane** [comment]
Package is commented out now and codacy issues fixed, but the check doesn't process now so I can't confirm that. Is codacy not reading it because it's not an active package?

File: packages/rocketchat-bot-helpers/server/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		});
+	}
+
+}
+
+// add class to meteor methods (BotHelper accepts options)
+// TODO: Let constructor options be passed in through another method (to override defaults)
+const botHelper = new BotHelper();
+Meteor.methods({
+	botRequest: (...args) => botHelper.request(...args)
```
Comment (rodrigok) [inline packages/rocketchat-bot-helpers/server/index.js]: Please, allow only logged users to call this method

File: packages/rocketchat-bot-helpers/server/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * BotHelper helps bots
+ * keeping Rocket.chat stuff in meteor package allows clean logic separation in bot scripts
+ * "private" properties use meteor collection cursors, so they stay reactive
+ * "public" "properties" use getters to fetch and filter collections as array
+ */
+class BotHelper {
+	constructor(options = {}) {
+		this.config = _.extend({}, defaults, options);
+		this._allUsers = Meteor.users.find(this.config.userQuery, { fields: this.config.userFields });
```
Comment (rodrigok) [inline packages/rocketchat-bot-helpers/server/index.js]: There is a security issue here, anyone can call this method and get all user's data, I can agree with some fields like name, username, utcOffset, etc.

**Comment by rodrigok** [comment]
@timkinnane do you need some help here?
Sorry by the delay, we was very busy with new amazing features and news 😄

**Comment by timkinnane** [comment]
No, thanks @rodrigok. I'm planning to update this by the end of the week, just haven't had time.

I appreciate your suggestions. Definitely worth adding more security. I plan to make the following changes:

- Allow only logged users with `bot` role to call methods
- Add admin page to configure which user fields can be accessed by bot requests

**Comment by rodrigok** [comment]
@timkinnane awesome, I agree with your suggestions and I will wait for theses changes then. 😄 

Thanks

**Comment by timkinnane** [comment]
@rodrigok All changes implemented.

The one thing I wasn't sure of is how to update the class property when the setting changes. Should I watch the setting directly in mongo, or is there some easier method for attaching callbacks to setting changes?

Alternatively I could get the setting on every method call so its always current, but that seems inefficient?

**Comment by rodrigok** [comment]
@timkinnane you can get the setting using callback `RocketChat.settings.get('BotHelpers_userFields', (value) => {})`, the callback will be called one time with the value and again for each update of the value.

**Comment by rodrigok** [comment]
@timkinnane let me know when I can review and merge this PR 😺

**Comment by timkinnane** [comment]
@rodrigok It's ready now! Just made final changes. I now have the admin setting available for changing which user fields the bot can access, with localized error strings when bot requests fields without permission.

**Comment by rodrigok** [comment]
@timkinnane awesome

**Pull request merged**